Add Attachment Management Plugin

trganda opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

Link to my plugin: [trganda/obsidian-attachment-management](https://github.com/trganda/obsidian-attachment-management)

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [x]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [ ] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
["description": "Attachment management plugin to custom the attachment path and auto rename etc.",](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/manifest.json#L5)
You should update your description we have recently published a guide for plugin descriptions to follow here: https://docs.obsidian.md/Plugins/Releasing/Submission+requirements+for+plugins#Keep+plugin+descriptions+short+and+simple.

[import * as path from "path";](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/src/main.ts#L3)
This is a Node.js library, it will break on mobile, as Node.js is not available there

.[this.adapter = this.app.vault.adapter as FileSystemAdapter;](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/src/main.ts#L39)
This will also break on mobile, `FileSystemAdapter` is only available on desktop.

[id: "attachment-management-override-setting",](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/src/main.ts#L55)
Obsidian will automatically prefix the command id with the plugin id,
you don't need to include it yourself.

[if the file is created more than 1 second ago, the event is most likely be fired on vault initialization when starting Obsidian app, ignore it](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/src/main.ts#L112)
Initialize your event handler inside `this.app.workspace.onLayoutReady()` instead.

[if (extension === "md" || extension === ".md") {](https://github.com/trganda/obsidian-attachment-management/blob/4eb1edeb4144ffe101ff386dbb65bc53025cebb8/src/utils.ts#L41)
You should be able to just remove the second condition, same below.

trganda commented:
I have updated the description.
["description": "Customize and your attachment path of notes with variables independently and auto rename it on change.",](https://github.com/trganda/obsidian-attachment-management/blob/main/manifest.json#LL5C122-L5C122)

`import * as path from "path";` was replaced with 
[a local path library](https://github.com/trganda/obsidian-attachment-management/blob/main/src/path.ts).

`this.adapter = this.app.vault.adapter as FileSystemAdapter` was replaced with `this.app.vault.adapter` directly.

removed the plugin id as prefix 
[id: "override-setting"](https://github.com/trganda/obsidian-attachment-management/blob/main/src/main.ts#LL53C20-L53C20)

second condition was removed 
[if (extension === "md")](https://github.com/trganda/obsidian-attachment-management/blob/main/src/utils.ts#L40)

trganda commented:
Hi @joethei.
I have made some modification, and you can chek it right now.

joethei commented:
The emulation is not perfect, some things cannot be emulated.

marcusolsson commented:
[name: "Reset Overrode Setting",](https://github.com/trganda/obsidian-attachment-management/blob/dbf50353eb53e0a1af9ebcb844e7666d155f6093/src/main.ts#L82)
Misspelled "Override".

[text: "Settings for Attachment Management.",](https://github.com/trganda/obsidian-attachment-management/blob/dbf50353eb53e0a1af9ebcb844e7666d155f6093/src/settings/settings.ts#L100)
https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#UI+text

trganda commented:
> [name: "Reset Overrode Setting",](https://github.com/trganda/obsidian-attachment-management/blob/dbf50353eb53e0a1af9ebcb844e7666d155f6093/src/main.ts#L82) Misspelled "Override".
> 
> [text: "Settings for Attachment Management.",](https://github.com/trganda/obsidian-attachment-management/blob/dbf50353eb53e0a1af9ebcb844e7666d155f6093/src/settings/settings.ts#L100) [docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#UI+text](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#UI+text)

Tks, I have fixed the typo error and removed the heading of setting

liamcain commented:
- [Rearrange Linked Attachments](https://github.com/trganda/obsidian-attachment-management/blob/35bd4890e28ffc51be3097e238184f4b90a9757b/src/main.ts#L41) similar to the settings menus, we also ask that plugins use sentence case for commands too so that the command palette feels consistent.

trganda commented:
> * [Rearrange Linked Attachments](https://github.com/trganda/obsidian-attachment-management/blob/35bd4890e28ffc51be3097e238184f4b90a9757b/src/main.ts#L41) similar to the settings menus, we also ask that plugins use sentence case for commands too so that the command palette feels consistent.

I have renamed command name to sentence case.
